packages/cli/src/util/link/handle-git-connection.ts

```diff
@@ -0,0 +1,271 @@
+    const currentProvider = project.link!.type;
+    output.log(
+      `Found Git remote url ${chalk.cyan(
+        remoteUrl
+      )}, which is different from the connected ${formatProvider(
+        currentProvider
+      )} repository ${chalk.cyan(currentRepoPath)}.`
+    );
+  } else {
+    output.log(`Found local Git remote URL: ${chalk.cyan(remoteUrl)}`);
+  }
+  const shouldConnect = await promptGitConnectSingleUrl(client, replace);
+  return handleOptions(
+    shouldConnect,
+    client,
+    output,
+    org,
+    project,
+    settings,
+    parsedUrl
+  );
+}
+
+async function addMultipleGitRemotes(
+  client: Client,
+  org: Org,
+  output: Output,
+  project: Project,
+  remoteUrls: Dictionary<string>,
+  settings: ProjectSettings
+) {
+  output.log('Found multiple Git remote URLs in Git config.');
+  const remoteUrlOrOptions = await promptGitConnectMultipleUrls(
+    client,
+    remoteUrls
+  );
+  return handleOptions(
+    remoteUrlOrOptions,
+    client,
+    output,
+    org,
+    project,
+    settings
+  );
+}
+
+async function handleOptions(
+  option: string,
+  client: Client,
+  output: Output,
+  org: Org,
+  project: Project,
+  settings: ProjectSettings,
+  parsedUrl?: {
+    provider: string;
+    org: string;
+    repo: string;
+  }
+) {
+  if (option === 'no') {
+    skip(output);
+  } else if (option === 'opt-out') {
+    await optOut(client, project, settings);
+  } else if (option !== '') {
+    // Option is "yes" or a URL
+    if (!parsedUrl) {
+      const _parsedUrl = parseRepoUrl(option);
+      if (!_parsedUrl) {
+        output.debug(`Could not parse repo url ${option}.`);
+        return;
+      }
+      parsedUrl = _parsedUrl;
+    }
+    const { provider, org: parsedOrg, repo } = parsedUrl;
+    const repoPath = `${parsedOrg}/${repo}`;
+
+    if (project.link) {
+      await disconnectGitProvider(client, org, project.id);
+    }
+    const connect = await connectGitProvider(
+      client,
+      org,
+      project.id,
+      provider,
+      repoPath
+    );
+    if (connect !== 1) {
+      output.log(
+        `Connected ${formatProvider(provider)} repository ${chalk.cyan(
+          repoPath
+        )}!`
+      );
+    } else {
+      return connect;
+    }
+  }
+}
+
+async function optOut(
+  client: Client,
+  project: Project,
+  settings: ProjectSettings
+) {
+  settings.skipGitConnectDuringLink = true;
+  await updateProject(client, project.name, settings);
+  client.output
+    .log(`Opted out. You can re-enable this prompt by visiting the Settings > Git page on the
+  dashboard for this Project.`);
+}
+
+function skip(output: Output) {
+  output.log('Skipping...');
+  output.log(
+    `You can connect a Git repository in the future by running ${getCommandName(
+      'git connect'
+    )}.`
+  );
+}
+
+export async function promptGitConnectSingleUrl(
+  client: Client,
+  replace = false
+) {
+  return await list(client, {
+    message: replace
+      ? 'Do you want to replace it?'
```

Comment:
What is "it" in this context? Can we print something more specific?

Reply:
This message appears right after another message that gives more context. What do you think of this?

<img width="683" alt="Screen Shot 2022-08-04 at 8 51 27 AM" src="https://user-images.githubusercontent.com/14811170/182891116-69e95ada-801e-43fe-98fa-73955f6e09cf.png">

May want to move the code around a little bit to make it more readable.

Reply:
Moved the logic that prints those messages next to each other so that it's clearer what's going on